src/torch/README.md

```diff
@@ -0,0 +1,123 @@
+# SIRF-PyTorch Wrapper
+This wrapper provides a bridge between the [SIRF](https://github.com/SyneRBI/SIRF) (Synergistic Image Reconstruction Framework) library and PyTorch, enabling the use of SIRF's image reconstruction operators and objective functions within PyTorch's automatic differentiation (autodiff) framework. 
+
+## Core Concepts: Autodiff, Reverse Mode, JVP, VJP, and HVP
+
+This wrapper leverages PyTorch's `torch.autograd.Function` to achieve integration with SIRF.  Understanding the following concepts is crucial for the design:
+
+*   **Automatic Differentiation (Autodiff):** A technique for numerically evaluating the derivative of a function specified by a computer program.  PyTorch uses a "reverse mode" autodiff system.
+
+*   **Reverse Mode Autodiff:**  In reverse mode, the gradients are computed by traversing the computational graph backwards, from the output (typically a scalar loss function) to the inputs.  This is highly efficient for functions with many inputs and a single output, which is common in machine learning.
+
+*   **Jacobian-Vector Product (JVP):**  Let `y = f(x)` be a vector-valued function, where `x` is an input vector and `y` is an output vector. The Jacobian matrix `J` contains all the first-order partial derivatives of `f`.  The JVP is the product `Jv`, where `v` is a vector.  It gives the directional derivative of `f` at `x` in the direction of `v`.  *Forward mode* autodiff computes JVPs.
+
+*   **Vector-Jacobian Product (VJP):** With `y = f(x)` as above, the VJP is the product `w^T J`, where `w` is a vector (often called the "upstream/puput gradient").  The VJP represents the gradient of a scalar-valued function that depends on `y` (e.g., a loss function), with respect to the input `x`.  *Reverse mode* autodiff computes VJPs. 
+
+*   **Hessian-Vector Product (HVP):** The Hessian matrix `H` contains all the second-order partial derivatives of a scalar-valued function `f(x)`. The HVP is the product `Hv`, where `v` is a vector.  It gives the second-order directional derivative of `f` at `x` in the direction of `v`. 
+
+## Wrapper Design
+
+The wrapper provides three main classes:
+
+1.  `SIRFTorchOperator`: Wraps a SIRF `Operator` (e.g., a projection operator). Computes the JVP forward and VJP in reverse.
+2.  `SIRFTorchObjectiveFunction`: Wraps a SIRF `ObjectiveFunction` for calculating its value forward, and gradient in reverse mode.
```

Comment:
it still computes VJP, i.e. whatever*gradient.

Reply:
for the objective function the gradient output usually just 1 (i.e. the seed for reverse mode autodiff) so that's why I omitted it, but it is more clear if I add this in